packages/core/src/application.ts

```diff
@@ -4,13 +4,34 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {Binding, Context, Constructor} from '@loopback/context';
+import {Component, Sequence} from '.';
 import * as http from 'http';
 import {SwaggerRouter} from './router/SwaggerRouter';
 import {getApiSpec} from './router/metadata';
 
 const debug = require('debug')('loopback:Application');
 
 export class Application extends Context {
+  constructor(public appConfig?: AppConfig) {
+    super();
+
+    if (appConfig && appConfig.components) {
+      for (const component of appConfig.components) {
+        // TODO(superkhau): Need to figure a way around this hack
+        const componentClassName = component.toString().split(' ')[1];
```

Comment:
FYI: The binding key ends up as `component.TodoComponent`, etc. We should discuss this during one of our sync ups as to keep uppercase/lowercase/cameCase/etc.